packages/components/src/local/organisms/setting-channels/helpers/generateRowItemsCollab.ts

```diff
@@ -0,0 +1,93 @@
+import { CollaborativePermission } from '@serge/config'
+import { ForceData, ParticipantCollab } from '@serge/custom-types'
+import { EDITABLE_SELECT_ITEM, EDITABLE_SWITCH_ITEM, Item, Option } from '../../../molecules/editable-row'
+import { Participant } from '../types/props'
+
+export default (forces: Array<ForceData>, nextParticipant: Participant): Array<Item> => {
+  // by default selected force
+  let forceSelected: Array<number> = [0]
+  // init empty roles array
+  let roleOptions: Array<Option> = []
+
+  // additional table cols for RFI participation toggles
+  const additionalFields: Item[] = []
+
+  if (nextParticipant.forceUniqid) {
+    // get selected force for current row
+    const forceIndex = forces.findIndex(force => force.uniqid === nextParticipant.forceUniqid)
+    if (forceIndex !== -1) {
+      // get all roles for current force and generate options for select
+      roleOptions = forces[forceIndex].roles.map((role): Option => ({
+        name: role.name,
+        uniqid: role.name,
+        value: role
+      }))
+      // selected value value for select
+      forceSelected = [forceIndex]
+    }
+  }
+
+  // get selected roles
+  const activeRoles: Array<number> = nextParticipant.roles.map(role => {
+    return roleOptions.findIndex(option => option.value.roleId === role)
+  }).filter(active => active !== -1)
+
+  // if this is a collaborative editing channel then the participant
+  // will be of a specific type
+  const collab = nextParticipant as unknown as ParticipantCollab
+
+  // init row item for create new message switch
+  additionalFields.push({
+    type: EDITABLE_SWITCH_ITEM,
+    uniqid: 'create_new_message',
+    // get default value for switcher
+    active: !!collab.canCreate
+  })
+
+  // init row item for see live updates switch
+  additionalFields.push({
+    type: EDITABLE_SWITCH_ITEM,
+    uniqid: 'see_live_updates',
+    // get default value for switcher
+    active: !!collab.viewUnreleasedVersions
+  })
+
+  const permissionOptions: Option[] = []
+  Object.keys(CollaborativePermission).forEach((key: number | string) => {
```

Comment:
I'm worried this this could be a number or a string. Doesn't our current implementation force the enum value to be a number?

Reply:
```
export enum CollaborativePermission {
  /** can */
  CannotCollaborate,
  CanEdit,
  CanSubmitForReview,
  CanApprove,
  CanRelease,
  CanUnClaim
}
```
this enum has type number @IanMayo :D

Reply:
```
export interface SelectItem {
  readonly uniqid: string
  type: typeof EDITABLE_SELECT_ITEM
  title?: string
  active?: Array<number>
  multiple?: boolean
  emptyTitle?: string
  options: Array<Option>
}
```
yep @IanMayo , the `active` items is an array of number.